### PR TITLE
did the task

### PR DESCRIPTION
**Pull request by vanyachyzh**
[DEMO LINK](https://vanyachyzh.github.io/react_todo-app-with-api/)

### CODE REVIEW

File: src/styles/todo.scss [hunk 1]
```diff
@@ -8,7 +8,7 @@
   font-size: 24px;
   line-height: 1.4em;
   border-bottom: 1px solid #ededed;
-
```
Comment (mykhalenych) [inline src/styles/todo.scss:11]: why did you add spaces between styles?

File: src/components/TodoItem/TodoItem.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+      return;
+    }
+
+    if (isEscape) {
+      changeTitle(event);
+    }
+  };
+
+  const onKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter' && isEmpty) {
```
Comment (mykhalenych) [inline src/components/TodoItem/TodoItem.tsx:79]: create enum with 'Enter' & 'Escape'

**Review by mykhalenych — CHANGES_REQUESTED** [review]
Screenshot 2023-03-24 at 15 39 33
after first click on button the status of the todos does not change, after the second click it changes

File: src/App.tsx [hunk 3]
```diff
@@ -1,24 +1,197 @@
...
+  };
+
+  const updateTodo = (id: number, data: Property) => {
+    setRemovedTodoId(id);
+    patchTodo(id, data)
+      .then((response) => {
+        const updatedTodos = [...todos];
+        const changedTodoId = [...todos].findIndex(todo => todo.id === id);
+
+        updatedTodos[changedTodoId] = response;
```
Comment (VitaliyBondarenko1982) [inline src/App.tsx:109]:
```suggestion
        const updatedTodos = todos.map(todo => todo.id === id ? response : todo);
```
it can be simplier

**Review by VitaliyBondarenko1982 — APPROVED** [review]
Good job.
